ecosystem/node-checker/DEV_NOTES.md

```diff
@@ -0,0 +1,41 @@
+Random thoughts while developing, I'll remove these before landing the stack.
```

Comment:
@JoshLind Not sure if you saw my PR disclaimer, I tried to address this hahah. I wrote this stack with the intent that each commit would be reviewed individually. I could make 20ish PRs but this feels like a lot of overhead. I also plan to generally keep my commit history clean such that I rebase, not squash. I was chatting with David recently and he was team rebase also (if your stack is clean enough, which this may not be). Perhaps we need to discuss as a team which approach to go with, I'm used to stack based workflows where every commit should pass tests, have a commit message, test plan, etc.

Reply:
I think 5 of your commits are something like `run cargo fmt` , `address lint`, `touch up X` , so those are not the kind of commits I'd like to see in main :)
In the current state I'd definitely squash this PR, but if you can reduce this to a few fewer commits it's fine rebase this instead.

One thing I wanted to call out the GitHub - for better or worse - is very PR oriented and the PR is the unit of change and contains most of the context.
I know at Meta you used phab and stacked diffs but GitHub just isn't really built with these workflows (such as stacking PRs) in mind and trying to bend GitHub to work like phabricator's review flow isn't trivial.

Actually on that note, if we were to get rid of bors/the merge queue, we could use something like this https://github.com/ejoffe/spr for people who like stacking PRs (I tried this out and it's pretty cool but it doesn't work with a merge queue). cc @davidiw  fyi

Reply:
Yeah this stack isn't really in a rebaseable state right now, agreed. I suppose committing to squashing does make my life easier because I don't have to make a clean stack, just a clean PR once it's all merged.

Reply:
> I wrote this stack with the intent that each commit would be reviewed individually

In practice I find it much quicker to review the entire stack at once (might be me though). Often times PR authors (myself included) will do some work in a commit, then undo parts of the very same work in the next commit with a slight modification, etc. This tends to lead unnecessary work (especially for super large PRs like this).

Reply:
Yeah sounds good, I realise with GitHub it's easier to treat one PR as one unit of review. My upcoming PRs are much smaller accordingly.